lib/Teng/Plugin/SearchJoined/Iterator.pm

```diff
@@ -43,9 +43,14 @@ sub _seperate_rows {
     my $name_sep = quotemeta $self->{teng}{sql_builder}{name_sep};
     my $i = 0;
     for my $field (@{ $self->{fields} }) {
-        my $value = $row->[$i++];
+        my $value = $row->[$i];
         my ($table, $column) = split /$name_sep/, $field;
+        if (!$column) {
+            $column = $self->{sth}->{NAME_lc}->[$i];
+            $table  = $self->{table_names}->[0]; # base_table
```

Comment:
I think it is not good idea using `base_table` as the table name. How about using '_' or so?